ServantHDBC/src/Api/Authentication.hs

```diff
@@ -0,0 +1,31 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Api.Authentication (AuthAPI, authHandlers) where
+
+import Control.Monad.Trans.Except
+import Servant
+
+type Auth = String
+type AuthAPI = "sessions" :>
+                         ("new" :> Post '[JSON] Auth
+                          :<|> "refresh" :> Post '[JSON] Auth
+                          :<|> "destroy" :> Post '[JSON] Auth)
+
+authNew :: ExceptT ServantErr IO Auth
```

Comment:
Yes, there are Header types available. I was putting that specific part off until i got to auth, i'd rather do it at that point.